subsys/bootloader/cmake/sign.cmake

```diff
@@ -196,3 +196,26 @@ foreach (slot ${slots})
     )
 
 endforeach()
+
+if (CONFIG_BUILD_S1_VARIANT AND NOT CONFIG_BOOTLOADER_MCUBOOT)
+  # B0 will boot the app directly, create the DFU zip. If MCUBoot is enabled,
+  # the DFU zip files is generated by the MCUBoot build code.
+
+  include(${NRF_DIR}/cmake/fw_zip.cmake)
+
+  # Generate zip file with both update candidates
+  set(s0_name signed_by_b0_s0_image.bin)
+  set(s0_bin_path ${PROJECT_BINARY_DIR}/${s0_name})
+  set(s1_name signed_by_b0_s1_image.bin)
+  set(s1_bin_path ${PROJECT_BINARY_DIR}/${s1_name})
+
+  generate_dfu_zip(
+    ${PROJECT_BINARY_DIR}/dfu_application.zip # Zip out
+    "${s0_bin_path};${s1_bin_path}"
+    "s0_image_signed_kernel_hex_target;s1_image_signed_kernel_hex_target" # dependencies
+    "type=application"
+    "${s0_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>"
+    "${s1_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>"
```

Comment:
Using generator expressions in strings this way is error prone.
it also puts restrictions on the `generate_dfu_zip` function in terms of refactoring or other modifications.

Reply:
The mechanism used here is that the python script interprets any argument starting with the name of the included file as parameters specific for that file. The alternative to this is to pass some data structure such as a map to the python script, which seems like overkill to me. 

Do you have any alternatives?

Reply:
The only alternative was to call `add_custom_command` directly, that would remove the need for putting the generator expression through a string, and later as argument to a function.